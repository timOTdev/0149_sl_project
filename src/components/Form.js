import React, { useState } from 'react'
import styled from 'styled-components'
import { ReactComponent as Arrow } from '../images/White_Arrow.svg'

const Form = () => {
  // HOOKS
  const [user, setUser] = useState({
    firstName: '',
    lastName: '',
    address1: '',
    address2: '',
  })

  // METHODS
  const handleUserChange = (e) => {
    e.preventDefault();
    setUser({
      ...user,
      [e.target.name]: e.target.value
    })
  }

  const handleUserSubmit = (e) => {
    e.preventDefault();
    const { firstName, lastName, address1, address2 } = user
    alert(`
      First Name: ${firstName},
      Last Name: ${lastName},
      Address1: ${address1},
      Address2: ${address2 !== '' ? address2 : ''}
    `)
    setUser({
      firstName: '',
      lastName: '',
      address1: '',
      address2: '',
    })
  }

  // RETURN
  return (
    <Section>
      <Form1 onSubmit={handleUserSubmit}>
        <Label htmlFor="firstName">FIRST NAME</Label>
        <Input
          name="firstName"
          type="text"
          value={user.firstName}
          onChange={handleUserChange}
          required
        />

        <Label htmlFor="lastName">LAST NAME</Label>
        <Input
          name="lastName"
          type="text"
          value={user.lastName}
          onChange={handleUserChange}
          required
        />

        <Label htmlFor="address1">ADDRESS</Label>
        <Input
          name="address1"
          type="text"
          value={user.address1}
          onChange={handleUserChange}
          required
        />

        <Label htmlFor="address2">ADDRESS 2 (OPTIONAL)</Label>
        <Input
          name="address2"
          type="text"
          value={user.address2}
          onChange={handleUserChange}
        />
        <Button type="submit">
          Next &nbsp; <Arrow />
        </Button>
      </Form1>
    </Section >
  )
}

const Section = styled.section`
  min-width: 32%;
  height: 100vh;
  background: #FFF;
`
const Form1 = styled.form`
  margin-top: 111px;
  margin-left: 24px;
  display: flex;
  flex-direction: column;
  width: 376px;
`
const Label = styled.label`
  color: #666666;
  font-family: 'Merriweather', 'serif';
  font-size: 1.2rem;
  text-align: left;
  margin-bottom: 8px;
`
const Input = styled.input`
  height: 48px;
  border-radius: 6px;
  margin-bottom: 24px;
`
const Button = styled.button`
  background: #FFAB44;
  color: white;
  border-radius: 6px;
  height: 48px;
  font-family: 'Roboto', 'sans-serif';
  font-size: 1.6rem;
  width: 106px;
  margin-top: 24px;
`

export default Form
