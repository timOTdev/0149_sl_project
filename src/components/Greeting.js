import React from 'react'
import styled from 'styled-components'
import { ReactComponent } from '../images/GiantRobotLTD_Logo.svg'

const Greeting = () => (
  <Section>
    <Logo1 />
    <H1>Welcome</H1>
    <P>Please tell us a bit about yourself to get started.</P>
  </Section>
)

const Section = styled.section`
  background: #585858;
  min-width: 57%;
  height: 100vh;
  text-align: left;
  padding-left: 170px;
  color: #FFFFFF;
`
const Logo1 = styled(ReactComponent)`
  margin-top: 32px;
  width: 177px;
  height: 26px;
`
const H1 = styled.h1`
  margin-top: 40px;
  font-family: 'Roboto', 'sans-serif';
  font-size: 3.2rem;
  margin-bottom: 8px;
`
const P = styled.p`
  font-family: 'Merriweather', 'serif';
  font-size: 1.4rem;
  color: #EAEAEA;
`

export default Greeting
