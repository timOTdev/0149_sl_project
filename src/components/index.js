// index for all component files
import Greeting from './Greeting'
import Form from './Form'

export {
  Greeting,
  Form
}
