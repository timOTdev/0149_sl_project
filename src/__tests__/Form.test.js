import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library'
import 'jest-dom/extend-expect'
import { Form } from '../components'

afterEach(cleanup)

describe('-- Form.js --', () => {
  it('renders', () => {
    const { container } = render(<Form />)
    expect(container).toMatchSnapshot()
  })

  describe('form element', () => {
    it('renders form element', () => {
      const { container } = render(<Form />)
      const formElement = document.querySelector('form')
      expect(container).toContainElement(formElement)
    })
  })

  describe('Input set 1', () => {
    it('renders label correctly', () => {
      const { getByTestId } = render(<Form />)
      expect(getByTestId('label1')).toHaveTextContent('FIRST NAME')
    })
    it('has attribute of required', () => {
      const { getByTestId } = render(<Form />)
      expect(getByTestId('input1')).toHaveAttribute('required')
    })
  })

  describe('Input set 2', () => {
    it('renders label correctly', () => {
      const { getByTestId } = render(<Form />)
      expect(getByTestId('label2')).toHaveTextContent('LAST NAME')
    })
    it('has attribute of required', () => {
      const { getByTestId } = render(<Form />)
      expect(getByTestId('input2')).toHaveAttribute('required')
    })
  })

  describe('Input set 3', () => {
    it('renders label correctly', () => {
      const { getByTestId } = render(<Form />)
      expect(getByTestId('label3')).toHaveTextContent('ADDRESS')
    })
    it('has attribute of required', () => {
      const { getByTestId } = render(<Form />)
      expect(getByTestId('input3')).toHaveAttribute('required')
    })
  })

  describe('Input set 4', () => {
    it('renders label correctly', () => {
      const { getByTestId } = render(<Form />)
      expect(getByTestId('label4')).toHaveTextContent('ADDRESS 2 (OPTIONAL)')
    })
  })

  describe('button element', () => {
    it('has type of submit', () => {
      render(<Form />)
      const buttonElement = document.querySelector('button')
      expect(buttonElement).toHaveAttribute('type', 'submit')
    })
  })
})
