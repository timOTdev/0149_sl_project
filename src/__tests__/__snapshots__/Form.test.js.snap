// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`-- Form.js -- renders 1`] = `
<div>
  <section
    class="sc-ifAKCX PvtxZ"
  >
    <form
      class="sc-EHOje KQgPY"
    >
      <label
        class="sc-bZQynM hLMmSv"
        data-testid="label1"
        for="firstName"
      >
        FIRST NAME
      </label>
      <input
        class="sc-gzVnrw HcCoo"
        data-testid="input1"
        name="firstName"
        required=""
        type="text"
        value=""
      />
      <label
        class="sc-bZQynM hLMmSv"
        data-testid="label2"
        for="lastName"
      >
        LAST NAME
      </label>
      <input
        class="sc-gzVnrw HcCoo"
        data-testid="input2"
        name="lastName"
        required=""
        type="text"
        value=""
      />
      <label
        class="sc-bZQynM hLMmSv"
        data-testid="label3"
        for="address1"
      >
        ADDRESS
      </label>
      <input
        class="sc-gzVnrw HcCoo"
        data-testid="input3"
        name="address1"
        required=""
        type="text"
        value=""
      />
      <label
        class="sc-bZQynM hLMmSv"
        data-testid="label4"
        for="address2"
      >
        ADDRESS 2 (OPTIONAL)
      </label>
      <input
        class="sc-htoDjs gehhcT"
        name="address2"
        placeholder=" "
        type="text"
        value=""
      />
      <button
        class="sc-dnqmqq bpbtYo"
        data-testid="button"
        type="submit"
      >
        Next
        <svg
          class="sc-iwsKbI eXVGwH"
        >
          White_Arrow.svg
        </svg>
      </button>
    </form>
  </section>
</div>
`;
